src/devices_models/devices/storage.jl

```diff
@@ -3,6 +3,7 @@
 abstract type AbstractStorageFormulation <: AbstractDeviceFormulation end
 struct BookKeeping <: AbstractStorageFormulation end
 struct BookKeepingwReservation <: AbstractStorageFormulation end
+struct EndOfPeriodEnergyTarget <: AbstractStorageFormulation end
```

Comment:
We should make some AbstractEnergyManagement that can enable other formulations with energy related constraints.